net/tsdial/tsdial.go

```diff
@@ -367,6 +367,8 @@ func (d *Dialer) PeerAPIHTTPClient() *http.Client {
 		t := http.DefaultTransport.(*http.Transport).Clone()
 		t.Dial = nil
 		t.DialContext = d.dialPeerAPI
+		// Do not use the environment proxy for peerapi.
+		t.Proxy = nil
```

Comment:
this is independently important, wondering how you caught it?

Reply:
see commit message